frontend/packages/kubevirt-plugin/src/components/create-vm/forms/boot-source-form.tsx

```diff
@@ -0,0 +1,325 @@
+  });
+
+  const useProjects = useFlag(FLAGS.OPENSHIFT);
+
+  const defaultSCName = getDefaultStorageClass(storageClasses)?.metadata.name;
+
+  const handleStorageClass = (scName: string) => {
+    const updatedStorageClass = storageClasses.find((sc) => sc.metadata.name === scName);
+    const provisioner = updatedStorageClass?.provisioner || '';
+    const modes: string[] =
+      provisionerAccessModeMapping[provisioner] || getAccessModeForProvisioner(provisioner);
+    dispatch({ type: BOOT_ACTION_TYPE.SET_ACCESS_MODE, payload: AccessMode.READ_WRITE_ONCE });
+    dispatch({
+      type: BOOT_ACTION_TYPE.SET_ACCESS_MODES,
+      payload: modes.map((m) => AccessMode.fromString(m)),
+    });
+    dispatch({
+      type: BOOT_ACTION_TYPE.SET_STORAGE_CLASS,
+      payload: updatedStorageClass?.metadata?.name,
+    });
+  };
+
+  if (scLoaded && !state.storageClass?.value) {
+    handleStorageClass(defaultSCName);
+  }
+
+  const isUpstream = window.SERVER_FLAGS.branding === 'okd';
+  return (
+    <Form onSubmit={preventDefault}>
+      <FormRow fieldId="form-data-source" title="Data source type" isRequired>
+        <FormPFSelect
+          placeholderText="--- Select data source ---"
+          aria-label="Select data source"
+          onSelect={(e, value) =>
+            dispatch({
+              type: BOOT_ACTION_TYPE.SET_DATA_SOURCE,
+              payload: value as DataSource,
+            })
+          }
+          selections={state.dataSource?.value}
+        >
+          {DataSource.getAll()
+            .filter((ds) => (withUpload ? true : ds !== DataSource.UPLOAD))
+            .map((ds) => (
+              <SelectOption key={ds.getValue()} value={ds} description={ds.getDescription()} />
+            ))}
+        </FormPFSelect>
+      </FormRow>
+      {state.dataSource?.value === DataSource.UPLOAD && (
+        <FormRow
+          fieldId="form-ds-file"
+          title="Upload source"
+          isRequired
+          validation={state.file?.validation}
+        >
+          <FileUpload
+            id="file-upload"
+            value={state.file?.value.value}
+            filename={state.file?.value.name}
+            onChange={(file: File, name: string) =>
+              dispatch({
+                type: BOOT_ACTION_TYPE.SET_FILE,
+                payload: { value: { value: file, name }, validation: undefined },
+              })
+            }
+            hideDefaultPreview
+            isRequired
+            dropzoneProps={{
+              accept: '.iso,.img,.qcow2,.gz,.xz',
+              onDropRejected: () =>
+                dispatch({
+                  type: BOOT_ACTION_TYPE.SET_FILE,
+                  payload: {
+                    value: {
+                      value: undefined,
+                      name: undefined,
+                    },
+                    validation: {
+                      message: 'Only .iso,.img,.qcow2,.gz,.xz are accepted',
+                      type: ValidationErrorType.Error,
```

Comment:
this should not be an error, see https://bugzilla.redhat.com/show_bug.cgi?id=1894374

Reply:
removed